src/ORM/DataObject.php

```diff
@@ -1319,7 +1319,14 @@ protected function onBeforeDelete()
 
         // Cascade deletes
         $deletes = $this->findCascadeDeletes(false);
+
+        /** @var DataObject $delete */
         foreach ($deletes as $delete) {
+            if (!$delete->canDelete()) {
```

Comment:
Probably just a preference thing as I'm not sure there's a convention for this, but I'd suggest using positively constructed conditions.
So here
```
// delete only models that give us permission to delete
if ($delete->canDelete()) {
   $delete->delete();
}
```
and otherwise do nothing.

Reply:
My preference is to use exit early approach to reduce code nesting first, but I generally agree that having positive conditions is a good thing. I don't have a strong opinion on this though. Happy to go either way ;-)

Reply:
Exit early is good if it helps avoid further computations or deeper nesting, here I guess it's 50/50 and anyone can argue both ways without having the upper hand :-)